core/application/src/query_runner.rs

```diff
@@ -82,198 +82,102 @@ impl QueryRunner {
-                .unwrap_or(None)
-        })
-    }
 }
 
 impl SyncQueryRunnerInterface for QueryRunner {
-    fn get_account_balance(&self, account: &EthAddress) -> u128 {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|a| a.bandwidth_balance)
-                .unwrap_or(0)
-        })
-    }
-    fn get_client_balance(&self, client: &ClientPublicKey) -> u128 {
-        self.inner.run(|ctx| {
-            let client_table = self.client_table.get(ctx);
-            let account_table = self.account_table.get(ctx);
-            // Lookup the account key in the client->account table and then check the balance on the
-            // account
-            client_table
-                .get(client)
-                .and_then(|key| account_table.get(key))
-                .map(|a| a.bandwidth_balance)
-                .unwrap_or(0)
-        })
+    fn get_metadata(&self, key: &Metadata) -> Option<Value> {
+        self.inner.run(|ctx| self.metadata_table.get(ctx).get(key))
     }
 
-    fn get_flk_balance(&self, account: &EthAddress) -> HpUfixed<18> {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|account| account.flk_balance)
-                .unwrap_or(HpUfixed::<18>::zero())
-        })
-    }
-
-    fn get_stables_balance(&self, account: &EthAddress) -> HpUfixed<6> {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|account| account.stables_balance)
-                .unwrap_or(HpUfixed::<6>::zero())
-        })
-    }
-
-    fn get_staked(&self, node: &NodePublicKey) -> HpUfixed<18> {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.staked)
-            .unwrap_or(HpUfixed::zero())
-    }
-
-    fn get_locked(&self, node: &NodePublicKey) -> HpUfixed<18> {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.locked)
-            .unwrap_or(HpUfixed::zero())
+    fn get_account_info<F: Clone>(
+        &self,
+        address: &EthAddress,
+        selector: impl FnOnce(AccountInfo) -> F,
+    ) -> Option<F> {
+        self.inner
+            .run(|ctx| self.account_table.get(ctx).get(address))
+            .map(selector)
     }
 
-    fn get_stake_locked_until(&self, node: &NodePublicKey) -> Epoch {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.stake_locked_until)
-            .unwrap_or(0)
+    fn get_client_info(self, pub_key: &ClientPublicKey) -> Option<EthAddress> {
+        self.inner
+            .run(|ctx| self.client_table.get(ctx).get(pub_key))
     }
 
-    fn get_locked_time(&self, node: &NodePublicKey) -> Epoch {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.locked_until)
-            .unwrap_or(0)
+    fn get_node_info<F: Clone>(
```

Comment:
```suggestion
    fn get_node_info<T>(
```
None of these returns should need to be Clone, 
Minor nit: Across all these functions lets rename the generic to T since its the return type of the closure and that is the typical naming convention. F makes me think it is the closure itself

Reply:
I can remove `Clone` requirement.
Regarding `F` -> `T` we cannot do that. I posted a message some time ago in Discord that with `#[infusion::blank]` trait we cannot use `T` parameter (probably it injects it somewhere). @daltoncoder if you wish I can change `F` to sth else.

Reply:
I think he was objecting to F because it's commonly used for closure bounds so I think it's fine to change it to anything else.

Reply:
Ah yeah i missed that message about infusion injecting T already. Let me work on getting a fix for that. In the meantime we can keep this F or whatever we want

Reply:
Changing to `V` then!